pkg/resource/role/hooks.go

```diff
@@ -342,6 +342,36 @@ func (rm *resourceManager) removeInlinePolicy(
 	return err
 }
 
+// syncAssumeRolePolicies syncs the role's assume role policy document.
+func (rm *resourceManager) syncAssumeRolePolicies(
+	ctx context.Context,
+	r *resource,
+) (err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.syncAssumeRolePolicies")
+	defer func() { exit(err) }()
+	return rm.putAssumeRolePolicies(ctx, r)
+}
+
+// putAssumeRolePolicies calls the IAM API to set a given role's
+// assume role policy document.
+func (rm *resourceManager) putAssumeRolePolicies(
+	ctx context.Context,
+	r *resource,
+) (err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.putAssumeRolePolicies")
+	defer func() { exit(err) }()
+
+	input := &svcsdk.UpdateAssumeRolePolicyInput{
+		RoleName:       r.ko.Spec.Name,
+		PolicyDocument: r.ko.Spec.AssumeRolePolicyDocument,
+	}
+	_, err = rm.sdkapi.UpdateAssumeRolePolicyWithContext(ctx, input)
+	rm.metrics.RecordAPICall("UPDATE", "UpdateAssumeRolePolicy", err)
+	return err
+}
```

Comment:
Agreed, I'll alter this now. My intention was to keep things aligned with the `syncX, putX, ...` styling, although this is probably where a diversion to that is completely acceptable.

> Plus, any chance we can add a simple test to test/e2e/tests/test_role.py verifying this behaviour?

Absolutely :rocket: I'm working on trying to piece this together now :blush: It's been awhile since I've done some Python, but I'm get the hang of it no doubt :laughing: